File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/UserRecord.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public class UserRecord {
    public UUID id;
    public String email;
    public String name;
    public String avatarUrl;
    public String googleSubjectId;
    public String gitHubSubjectId;
    public Instant createdAt;
    public Instant lastLoginAt;
    public String displayName;
    public String timezone;
    public String notificationPrefsJson;
    public String preferences;
    public UserSubscription subscription;
}

File: apps/backend/src/Features/Management/Domain/UserRecord.cs
namespace OffceOs.Domain.Features.Management;

public sealed class UserRecord
{
    public Guid Id { get; init; } = Guid.NewGuid();
    [Required, MaxLength(256)]
    public string Email { get; init; } = string.Empty;
    [MaxLength(256)]
    public string? Name { get; set; }
    [MaxLength(1024)]
    public string? AvatarUrl { get; set; }
    [MaxLength(256)]
    public string? GoogleSubjectId { get; init; }
    [MaxLength(256)]
    public string? GitHubSubjectId { get; init; }
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime LastLoginAt { get; set; } = DateTime.UtcNow;

    // Profile fields — editable via AuthMutations.updateProfile
    [MaxLength(128)]
    public string? DisplayName { get; set; }
    [MaxLength(64)]
    public string? Timezone { get; set; }
    /// <summary>JSON blob of notification preferences, e.g. {"taskCompletions":true,"email":false,"channelMessages":true}.</summary>
    public string? NotificationPrefsJson { get; set; }

    /// <summary>Free-text personal preferences that apply to all agents (e.g. "keep explanations brief").</summary>
    [MaxLength(4000)]
    public string? Preferences { get; set; }

    public UserSubscription? Subscription { get; init; }
}
